dispatcher/msgdispatcher_test.go

```diff
@@ -561,4 +561,41 @@ func TestJobWorkers(t *testing.T) {
 			}
 		}
 	})
+	t.Run("SuccessRecoverLongInflightWithTimeout", func(t *testing.T) {
+		dispatcher := NewMessageDispatcher(getCompleteDispatcherConfiguration(dataAccessor.GetMessageRepository(), dataAccessor.GetDeliveryJobRepository(), dataAccessor.GetConsumerRepository(), brokerConf, configuration, dataAccessor.GetLockRepository()))
+		impl := dispatcher.(*MessageDispatcherImpl)
+
+		var inflightPullJob *data.DeliveryJob = nil
+		for _, job := range jobs {
+			if job.Listener.Type == data.PullConsumer {
```

Comment:
You should also have a test for `PushConsumer`

Reply:
Done bhaia.